source/module_hsolver/hsolver.h

```diff
@@ -59,6 +59,7 @@ class HSolver
         psi::Psi<std::complex<double>>& ppsi, 
         elecstate::ElecState* pes, 
         Stochastic_WF& stowf,
+        const int istep,
```

Comment:
In my opinion, interface in base class should only has input parameters which is general for every derived class, if there is any special demand in some derived class, developer should find someway else to pass their parameters rather than add  them into virtual interface.

Reply:
I think it is essential for parameter "iter" and "istep." in hamilt2density. Since it locates in the loop of istep and iter.

Reply:
It is not good to hide them. It will make codes hard to read.